### PR TITLE
Refactor Orders page according to new requirements #396

### PR DESCRIPTION
**Pull request by dmytro07**
Implemented possibility to create new orders from admin panel
 Implemented possibility to delete items from order
 Implemented possibility to add items to order
 Implemented possibility edit non fixed fields of order (delivery, user info, statuses, payment, cancelation)

Screenshot from 2021-03-22 22-27-41
Screenshot from 2021-03-22 22-27-43
Screenshot from 2021-03-22 22-27-47
Screenshot from 2021-03-22 22-27-49

### Checklist
- [ ] 🔽 My branch is up-to-date with "development" branch
- [ ] ✅ All tests passed locally
- [ ] ✨ My changes working with up-to-date front-end and back-end part locally, like charm
- [ ] 🔗 Link pull request to issue

### CODE REVIEW

File: src/pages/order-item/order-item.js [hunk 1]
```diff
@@ -110,30 +99,34 @@ const OrderItem = ({ id }) => {
...
-          color='primary'
-          className={classes.saveBtn}
-        >
-          {SAVE_ORDER}
-        </Button>
-      )}
+      <Button
+        type='submit'
+        variant='contained'
+        color='primary'
```
Comment (AndriyLymych) [inline src/pages/order-item/order-item.js:111]: please,dont use magic strings

File: src/pages/order-item/tabs/add-product-form/add-product-form.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+          disabled={!selectedProduct}
+          value={size}
+          onChange={selectHandler}
+        >
+          {sizeItems}
+        </Select>
+      </div>
+      <Button
+        variant='contained'
+        color='primary'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/add-product-form/add-product-form.js:134]: same

File: src/pages/order-item/tabs/delivery-details/nova-post/nova-post.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+              (warehouseItem) => !warehouseItem.description.includes(POSTOMAT)
+            )}
+            inputValue={departmentFocus ? wareHouse : values.courierOffice}
+            getOptionLabel={(option) => option?.description}
+            className={styles.dataInput}
+            renderInput={(params) => (
+              <TextField
+                {...params}
+                label={deliveryLabels.department}
+                variant='outlined'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery-details/nova-post/nova-post.js:123]: same

File: src/pages/order-item/tabs/delivery-details/nova-post/nova-post.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+            }}
+            options={cities}
+            inputValue={cityFocus ? inputValue : values.city}
+            getOptionLabel={(option) => option?.description || null}
+            className={styles.dataInput}
+            renderInput={(params) => (
+              <TextField
+                {...params}
+                label={deliveryLabels.city}
+                variant='outlined'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery-details/nova-post/nova-post.js:76]: same

File: src/pages/order-item/tabs/delivery-details/ukrpost/ukrpost.js [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+          }}
+          options={ukrPoshtaRegions}
+          inputValue={region}
+          getOptionLabel={(option) => option?.REGION_UA || ''}
+          className={styles.dataInput}
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              label={deliveryLabels.region}
+              variant='outlined'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery-details/ukrpost/ukrpost.js:104]: same

File: src/pages/order-item/tabs/delivery-details/ukrpost/ukrpost.js [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+          disabled={!region}
+          options={ukrPoshtaDistricts}
+          inputValue={district}
+          getOptionLabel={(option) => option?.DISTRICT_UA || ''}
+          className={styles.dataInput}
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              label={deliveryLabels.district}
+              variant='outlined'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery-details/ukrpost/ukrpost.js:142]: same

File: src/pages/order-item/tabs/delivery-details/ukrpost/ukrpost.js [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+            `${POST_OFFICE_NUMBER} ${option?.POSTCODE}, ${
+              option?.STREET_UA_VPZ ? option?.STREET_UA_VPZ : ''
+            }` || ''
+          }
+          className={styles.dataInput}
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              label={deliveryLabels.department}
+              variant='outlined'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery-details/ukrpost/ukrpost.js:214]: same

File: src/pages/order-item/tabs/recipient.js [hunk 8]
```diff
@@ -1,34 +1,38 @@
...
   return (
     <div className={classes.recipient}>
-      {Object.keys(user).map((item) => (
+      {user &&
+        Object.keys(user).map((item) => (
+          <TextField
+            name={`user.${item}`}
+            label={orderRecipient[item] || ''}
+            key={item}
+            variant='outlined'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/recipient.js:20]: same

File: src/redux/orders/orders.operations.js [hunk 9]
```diff
@@ -179,6 +179,23 @@ export const updateOrder = (order, id) => {
...
+				...on Order {
+					orderNumber
+				}
+				...on Error {
+					statusCode
+					message
+				}
+			}
+		}
+  `;
```
Comment (AndriyLymych) [inline src/redux/orders/orders.operations.js:195]: please, pretify this

**Comment by Kiyooooko** [comment]
pls make git pull origin development

**Review by AndriyLymych — DISMISSED** [review]

**Review by IlyaBelik — DISMISSED** [review]

**Comment by grebiniuk** [comment]
These fields should not be empty after editing or adding new order:
image

**Review by grebiniuk — APPROVED** [review]

**Review by AndriyLymych — APPROVED** [review]

**Review by IlyaBelik — APPROVED** [review]

**Comment by dmytro07** [comment]
https://github.com/orgs/ita-social-projects/projects/6#card-57607436

**Pull request merged**